src/Wallabag/CoreBundle/Operator/Doctrine/Matches.php

```diff
@@ -0,0 +1,15 @@
+<?php
+
+namespace Wallabag\CoreBundle\Operator\Doctrine;
+
+class Matches
+{
+    public function __invoke($subject, $pattern)
+    {
+        if ($pattern[0] === "'") {
+            $pattern = sprintf("'%%%s%%'", substr($pattern, 1, -1));
+        }
+
+        return sprintf('UPPER(%s) LIKE UPPER(%s)', $subject, $pattern);
```

Comment:
Is it working on MySQL, PostGres & SQLite ? Maybe a working test for the matches keyword could be a good idea.